### DISCUSSION TITLE
OF-8

### DISCUSSION THREAD

**Discussion by Parsa-Aalto** [WENO-OF/WENOEXT#13] [General]
Hi, 

I hope that you are doing very well.
Could I kindly ask whether you have any near future plan to release the code for OF-8? 

Regards,

**Comment by JanGaertner** [comment]
** I just tested it and it compiles with OpenFOAM 8. What issue do you have? ** 

Hello,

yes I do, I am currently working on a larger update that should include a better testing framework and solve some issues, as well as integrating cyclic boundaries. 
Also code is cleaned up and I aim for a better documentation.

However, a bigger problem is the stencil collection on structured grids and the condition of the pseudo inverse matrices. If you or someone can offer help to resolve this issues I would be happy to implement this. *Note:* The strategie of the stencil based compact algorithm presented by Tsoutsanis is already implemented.

**Reply by Parsa-Aalto** [reply]
Thank you so much Jan for your reply.
I could compile it on OF-7 but when it comes to OF-8, it gives me the following error:
error: 'const class Foam::cyclicPolyPatch' has no member named 'neighbPatchID'; did you mean 'nbrPatchID'?

Thank you so much for the update.

**Reply by JanGaertner** [reply]
Hello,

I have uploaded a new branch, dev-3.x, which is my current development branch for the next release. This one can also build for OpenFOAM 8. The reason why it did not work before, is that OpenFOAM changed the names of their coupled patches from OF7 to OF8 (see the release notes: [here](https://openfoam.org/release/8/))

One important change from version 2.x to 3.x is the use of cmake to compile the library. If you use the Allwmake script nothing changes.

**Reply by Parsa-Aalto** [reply]
Hi,

Thank you so much Jan.
I will check it and will let you know if I face any problem.

Regards,
